src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -757,16 +754,10 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
 
 #endif
             }
-            else if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-            {
-                ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
-            }
 
-            GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL;
-            if (pinned)
-                flags |= GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
+            GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL | GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
```

Comment:
```suggestion
            GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL;
            
            if (pinned)	
                flags |= GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
```
We do not want to be pinning the array unconditionally. This should be kept intact.

Reply:
Yeah, last commit was rushed. Please check latest (should fix below as well).